main/src/main/scala/sbt/internal/server/BuildServerProtocol.scala

```
@@ -186,6 +186,23 @@ object BuildServerProtocol {
     bspBuildTargetCompile / aggregate := false,
     bspBuildTargetTest := bspTestTask.evaluated,
     bspBuildTargetTest / aggregate := false,
+    bspBuildTargetCleanCache := Def.inputTaskDyn {
+      val s: State = state.value
+      val targets = spaceDelimited().parsed.map(uri => BuildTargetIdentifier(URI.create(uri)))
+      val workspace = bspFullWorkspace.value.filter(targets)
+      workspace.warnIfBuildsNonEmpty(Method.Compile, s.log)
+      val filter = ScopeFilter.in(workspace.scopes.values.toList)
+      Def.task {
+        val statusCodes = Keys.clean.result.all(filter).value
+        val statusCode = allOrThrow(statusCodes) match {
+          case Seq() => false
+          case _     => true
+        }
```

The protocol is not really clear but I don't think we should throw an exception if only one build target fails. So I suggest this instead.

```scala
      val result = anyOrThrow(statusCodes).size == targets.size
```

If all the targets fail we throw an exception otherwise we check that the number of successful targets is equal to the number of targets in the request.

If `result` is false, that means there is at least one target that failed. If it is true everything is okay.